File: 기초 사고력문제/백준_2798_블랙잭/블랙잭.cpp
#include <iostream>

using namespace std;

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    int arr[100] ={0};

    for(int i = 0; i < n; i++) scanf("%d", &arr[i]);

    int max = 0;
    for(int i = 0; i < n-2; i++){
        for(int j = i+1; j < n-1; j++){
            for(int k = j+1; k < n; k++){
                int sum = arr[i] + arr[j] + arr[k];
                if(sum <= m){
                    if(sum > max) max = sum;
                }
            }
        }
    }
    printf("%d", max);
}

File: 기초 사고력문제/백준_2798_블랙잭/블랙잭.py
n,m = map(int, input().split(' '))
l = list(map(int, input().split(' ')))

max = 0
for i in range(0,len(l) - 2):
    for j in range(i+1, len(l) - 1):
        for k in range(j+1, len(l)):
            sum = l[i] + l[j] + l[k]
            if sum <= m:
                if max < sum : max = sum
print(max)